auth/presentation/src/main/java/com/gkp/auth/presentation/login/LoginScreen.kt

```diff
@@ -0,0 +1,145 @@
+package com.gkp.auth.presentation.login
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeContentPadding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.input.rememberTextFieldState
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.LinkAnnotation
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextLinkStyles
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.withLink
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewLightDark
+import androidx.compose.ui.unit.dp
+import com.gkp.auth.presentation.R
+import com.gkp.auth.presentation.components.AuthBlackBackground
+import com.gkp.auth.presentation.components.PrimaryButton
+import com.gkp.auth.presentation.components.TaskyTextField
+import com.gkp.core.designsystem.theme.Inter
+import com.gkp.core.designsystem.theme.TaskyTextFieldColor
+import com.gkp.core.designsystem.theme.TaskyTheme
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+internal fun LoginScreen(
+    modifier: Modifier = Modifier,
+    onLoginButtonClick: () -> Unit,
+    onSignUpTextClick: () -> Unit
+) {
+    val viewModel = koinViewModel<LoginViewModel>()
+
+    LoginScreen(
+        onLoginButtonClick = onLoginButtonClick,
+        onSignUpTextClick = onSignUpTextClick,
+    )
+}
+
+@Composable
+private fun LoginScreen(
+    onLoginButtonClick: () -> Unit,
+    onSignUpTextClick: () -> Unit
+) {
+    val emailTextState = rememberTextFieldState()
+    val passwordTextState = rememberTextFieldState()
```

Comment:
Personally, I'd keep all states part of the ViewModel, since managing and updating that is the job of the ViewModel. If you have some state in the UI and some in the ViewModel, the code lacks consistency and it's not clear what to find where. It can just be part of your UI state like that:
```
data class LoginState(
    val emailTextState: TextFieldState = TextFieldState(),
    ...
)
```

Reply:
You' re right. I move then to viewModel when it will starts its implementation